addons/web/controllers/binary.py

```diff
@@ -84,32 +84,63 @@ def content_common(self, xmlid=None, model='ir.attachment', id=None, field='raw'
         res.headers['Content-Security-Policy'] = "default-src 'none'"
         return res
 
-    @http.route(['/web/assets/debug/<string:filename>',
-        '/web/assets/debug/<path:extra>/<string:filename>',
-        '/web/assets/<int:id>/<string:filename>',
-        '/web/assets/<int:id>-<string:unique>/<string:filename>',
-        '/web/assets/<int:id>-<string:unique>/<path:extra>/<string:filename>'], type='http', auth="public")
-    # pylint: disable=redefined-builtin,invalid-name
-    def content_assets(self, id=None, filename=None, unique=False, extra=None, nocache=False):
-        if not id:
-            domain = [('url', '!=', False)]
-            if extra:
-                domain += [('url', '=like', f'/web/assets/%/{extra}/{filename}')]
-            else:
-                domain += [
-                    ('url', '=like', f'/web/assets/%/{filename}'),
-                    ('url', 'not like', f'/web/assets/%/%/{filename}')
-                ]
+    @http.route([
+        '/web/assets/<path:extra>/<string:filename>',
+        '/web/assets/<string:unique>/<path:extra>/<string:filename>'], type='http', auth="public")
+    def content_assets(self, filename=None, unique='%', extra='-', nocache=False):
+        debug_assets = unique == 'debug'
+        attachment = None
+        if unique != 'debug':
+            url = f'/web/assets/{unique}/{extra}/{filename}'
+            domain = [
+                ('public', '=', True),
+                ('url', '!=', False),
+                ('url', '=like', url),
+                ('res_model', '=', 'ir.ui.view'),
+                ('res_id', '=', 0),
+                ('create_uid', '=', SUPERUSER_ID),
+            ]
             attachment = request.env['ir.attachment'].sudo().search(domain, limit=1)
-            if not attachment:
-                raise request.not_found()
-            id = attachment.id
-        with replace_exceptions(UserError, by=request.not_found()):
-            record = request.env['ir.binary']._find_record(res_id=int(id))
-            stream = request.env['ir.binary']._get_stream_from(record, 'raw', filename)
-
+        if not attachment:
+            # try to generate one
+            with replace_exceptions(ValueError, by=werkzeug.exceptions.BadRequest):
+                if debug_assets:
+                    bundle_name, asset_type = filename.rsplit('.', 1)
+                else:
+                    bundle_name, _min, asset_type = filename.rsplit('.', 2)
```

Comment:
Style nitpicking: `_min` is typically for a variable that you don't use.  What about `min_` instead?